include/grpc/grpc_security.h

```diff
@@ -143,6 +143,16 @@ grpc_channel_credentials *grpc_google_default_credentials_create(void);
 #define GRPC_DEFAULT_SSL_ROOTS_FILE_PATH_ENV_VAR \
   "GRPC_DEFAULT_SSL_ROOTS_FILE_PATH"
 
+/* Overrides the default path for TLS/SSL roots.
+   The path must point to a PEM encoded file with all the roots such as the one
+   that can be downloaded from https://pki.google.com/roots.pem.
+   This function is not thread-safe and must be called at initialization time
+   before any ssl credentials are created to have the desired side effect.
+   It also does not do any checks about the validity or contents of the path.
+   If the GRPC_DEFAULT_SSL_ROOTS_FILE_PATH environment is set, it will override
+   the roots_path specified in this function. */
+void grpc_override_ssl_default_roots_file_path(const char *roots_path);
```

Comment:
Right. Windows (at least) has a way of storing data alongside code that's
not in the file system.

On Wed, Jan 27, 2016 at 3:58 PM jboeuf <notifications@github.com> wrote:

> In include/grpc/grpc_security.h
> <https://github.com/grpc/grpc/pull/4929#discussion_r51070613>:
>
> > @@ -143,6 +143,16 @@ grpc_channel_credentials *grpc_google_default_credentials_create(void);
> >  #define GRPC_DEFAULT_SSL_ROOTS_FILE_PATH_ENV_VAR \
> >    "GRPC_DEFAULT_SSL_ROOTS_FILE_PATH"
> >
> > +/* Overrides the default path for TLS/SSL roots.
> > +   The path must point to a PEM encoded file with all the roots such as the one
> > +   that can be downloaded from https://pki.google.com/roots.pem.
> > +   This function is not thread-safe and must be called at initialization time
> > +   before any ssl credentials are created to have the desired side effect.
> > +   It also does not do any checks about the validity or contents of the path.
> > +   If the GRPC_DEFAULT_SSL_ROOTS_FILE_PATH environment is set, it will override
> > +   the roots_path specified in this function. */
> > +void grpc_override_ssl_default_roots_file_path(const char *roots_path);
>
> so that would be more:
>
> void grpc_override_ssl_default_roots(const char *roots_pem);
>
> i..e it's not a file anymore, it's the pem string directly, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/4929/files#r51070613>.
>